Sources/GRPCCodeGen/Internal/StructuredSwiftRepresentation.swift

```diff
@@ -483,7 +488,13 @@ struct ParameterDescription: Equatable, Codable {
   /// The type name of the parameter.
   ///
   /// For example, in `bar baz: String = "hi"`, `type` is `String`.
-  var type: ExistingTypeDescription
+  var type: ExistingTypeDescription? = nil
+
+  /// The description of the closure parameter.
+  ///
+  /// The parameter doesn't have a type, but is a closure.
+  /// For example: `(String) async throws -> Int`.
+  var closureDescription: ClosureSignatureDescription? = nil
```

Comment:
I have decided to have a different variable that signals if the parameter is a closure (and also its description) because the ExistingTypeDescription rendering doesn't allow more complex types.